-added the following models:

YoadK opened the pull request:
*credentials.model.ts
*user-role.enum.ts
*user.model.ts

-Reorganized the 'server/src' directory for improved structure.

YoadK committed:
-added the following models:
*credentials.model.ts
*user-role.enum.ts
*user.model.ts

-Reorganized the 'server/src' directory for improved structure.

benshabbat commented on server/package.json:
```diff
@@ -18,17 +18,13 @@
     "typescript": "^5.8.3"
   },
   "dependencies": {
-    "@types/bcrypt": "^5.0.2",
     "@types/cors": "^2.8.17",
     "@types/express": "^5.0.1",
     "@types/jsonwebtoken": "^9.0.9",
-    "@types/mongoose": "^5.11.96",
-    "bcrypt": "^5.1.1",
-    "cookie-parser": "^1.4.7",
     "cors": "^2.8.5",
     "dotenv": "^16.5.0",
     "express": "^5.1.0",
     "jsonwebtoken": "^9.0.2",
-    "mongoose": "^8.14.0"
```
you cant to un install packages.

benshabbat commented on server/package-lock.json:
you cant touch in this file

benshabbat commented on server/src/models/user.model.ts:
```diff
@@ -1,19 +0,0 @@
-    phone: { type: String, required: true },
-    password: { type: String, required: true },
-    role: { type: String, required: true, enum: ["admin", "user"] },
-  },
-  {
-    timestamps: true,
-  }
-);
```
add timestamp

benshabbat commented on server/src/middleware/user.middleware.ts:
dont delete this one

benshabbat commented on server/src/models/user.model.ts:
```diff
@@ -1,19 +0,0 @@
-
-const UserSchema: Schema = new Schema(
-  {
-    firstName: { type: String, required: true },
-    lastName: { type: String, required: true },
-    email: { type: String, required: true, unique: true },
-    phone: { type: String, required: true },
-    password: { type: String, required: true },
-    role: { type: String, required: true, enum: ["admin", "user"] },
```
i want same types and names

benshabbat commented on server/src/interfaces/user.interface.ts:
```diff
@@ -1,8 +0,0 @@
-export interface IUser {
-    firtsName: string;
-    lastName: string;
-    phone: string;
-    email: string;
-    password: string;
-    role: string;
-}
```
i want same types and names

benshabbat commented on server/src/6-controllers/user.controller.ts:
```diff
@@ -0,0 +1 @@
+import UserModel from "../3-models/user.model";
```
please without numbers in folder

benshabbat commented on server/src/3-models/user.model.ts:
```diff
@@ -0,0 +1,21 @@
+import { Document, ObjectId, Schema, model } from 'mongoose';
+
+export interface IUser extends Document {
+    _id: ObjectId;
+  firstName: string;
+  lastName: string;
+  email: string;
+  password: string;
+  roleId: number;
+}
+
+const UserSchema = new Schema<IUser>({
+  firstName: { type: String, required: true },
+  lastName: { type: String, required: true },
+  email: { type: String, required: true, unique: true },
+  password: { type: String, required: true },
+  roleId: { type: Number, required: true }
+});
+
+export const UserModel = model<IUser>('User', UserSchema, 'users');
+
```
fix same types and names

benshabbat commented on server/src/3-models/credentials.model.ts:
```diff
@@ -0,0 +1,26 @@
+import { Schema } from "mongoose";
+
+export interface ICredentials {
+    email: string;
+    password: string;
+  }
+
+  // Schema for ICredentialsModel
+export const CredentialsSchema = new Schema<ICredentials>({
+    email: {
+      type: String,
+      required: [true, "Missing email"],
+      unique: true,
+      minlength: [2, "email name is too short"],
+      maxlength: [50, "email name is too long"],
+    },
+    password: {
+      type: String,
+      required: [true, "Missing password"],
+      minlength: [4, "password  is too short"],
+      maxlength: [50, "password  is too long"],
+    },
+  }, {
+    versionKey: false,
+  });
+  
```
i want issue for this one

YoadK committed:
-removed 'credentials model'
- fixed role type
- fixed package.json dependencies
- re-installed bcryptjs, cookie-parser
-user model: added the timestamp option.
-properly re-named folder names (removed numbering)

benshabbat commented on server/package-lock.json:
pls back this package.

YoadK committed:
updated packages using npm update.
package-lock.json exists.

benshabbat commented on server/package-lock.json:
pls take the file from main no to create new one.. do with merge
